sdks/go/test/load/util.go

```diff
@@ -0,0 +1,227 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package load
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/apache/beam/sdks/go/pkg/beam"
+	"github.com/apache/beam/sdks/go/pkg/beam/core/metrics"
+)
+
+const (
+	runtimeMetricNamespace = "RuntimeMonitor"
+	runtimeMetricName      = "runtime"
+)
+
+var (
+	influxMeasurement = flag.String(
+		"influx_measurement",
+		"",
+		`An InfluxDB measurement where metrics should be published to.
+		If empty, no metrics will be send to InfluxDB.`)
+	influxDatabase = flag.String(
+		"influx_db_name",
+		"",
+		"InfluxDB database name. If empty, no metrics will be send to InfluxDB.")
+	influxHost = flag.String(
+		"influx_hostname",
+		"http://localhost:8086",
+		"Hostname and port to connect to InfluxDB. Defaults to http://localhost:8086.")
+	influxNamespace = flag.String(
+		"influx_namespace",
+		"",
+		`A namespace to be used when constructing InfluxDB's data points.
+		Used to make some points different from others within the same measurement.`)
+	runtime = beam.NewDistribution(runtimeMetricNamespace, runtimeMetricName)
+)
+
+func init() {
+	beam.RegisterType(reflect.TypeOf((*RuntimeMonitor)(nil)).Elem())
+}
+
+// RuntimeMonitor is a DoFn to record processing time in the pipeline.
+//
+// It uses a distribution metric which is updated every time new bundle
+// starts or finishes. The processing time can be extracted by calculating
+// the difference of the maximum and the minimum value of the distribution
+// metric.
+type RuntimeMonitor struct{}
+
+// StartBundle updates a distribution metric.
+func (fn *RuntimeMonitor) StartBundle(ctx context.Context, emit func([]byte, []byte)) {
+	runtime.Update(ctx, time.Now().Unix())
```

Comment:
Note, Unix only provides second granularity. If seconds is too coarse (or doesn't match Java and Python), consider using UnixNano() and converting to the correct granularity (millis or micros).

Reply:
(Notionally I think this is the only thing I need confirmation on before merging this)

Reply:
I think there's no reason we can't use `UnixNano()`. The database expects the number of seconds, but the fractional part can provide valuable information, especially for small tests.

Reply:
SGTM As long as it ends up with the right units for the DB vs Java and Python, it's fine. I'm less familiar so I must ask :).